doc/source/intro.rst

```diff
@@ -0,0 +1,136 @@
+
+Gnocchi can leverage different storage systems, such as:
+
+* File (default)
+* `Ceph`_ (preferred)
+* `OpenStack Swift`_
+* `Amazon S3`_
+* `Redis`_
+
+Depending on the size of your architecture, using the file driver and storing
+your data on a disk might be enough. If you need to scale the number of server
+with the file driver, you can export and share the data via NFS among all
+Gnocchi processes. In any case, it is obvious that S3, Ceph and Swift drivers
+are largely more scalable. Ceph also offers better consistency, and hence is
+the recommended driver.
+
+.. _`OpenStack Swift`: http://docs.openstack.org/developer/swift/
+.. _`Ceph`: https://ceph.com
+.. _`Amazon S3`: https://aws.amazon.com/s3/
+.. _`Redis`: https://redis.io
+
+Index driver
+~~~~~~~~~~~~
+
+You also need a database to index the resources and metrics that Gnocchi will
+handle. The supported drivers are:
+
+* `PostgreSQL`_ (preferred)
+* `MySQL`_ (at least version 5.6.4)
+
+The *indexer* is responsible for storing the index of all |resources|, |archive
+policies| and |metrics|, along with their definitions, types and properties.
+The indexer is also responsible for linking |resources| with |metrics|.
+
+.. _PostgreSQL: http://postgresql.org
+.. _MySQL: http://mysql.org
+
+Architecture overview
+---------------------
+
+Gnocchi consists of several services: a HTTP REST API (see :doc:`rest`), an
+optional statsd-compatible daemon (see :doc:`statsd`), and an asynchronous
+processing daemon (named `gnocchi-metricd`). Data is received via the HTTP REST
+API or statsd daemon. `gnocchi-metricd` performs operations (statistics
+computing, |metric| cleanup, etc...) on the received data in the background.
+
+.. image:: _static/architecture.svg
+  :align: center
+  :width: 95%
+  :alt: Gnocchi architecture
+
+.. image source: https://docs.google.com/drawings/d/1aHV86TPNFt7FlCLEjsTvV9FWoFYxXCaQOzfg7NdXVwM/edit?usp=sharing
+
+All those services are stateless and therefore horizontally scalable. Contrary
+to many time series databases, there is no limit on the number of
+`gnocchi-metricd` daemons or `gnocchi-api` endpoints that you can run. If your
+load starts to increase, you just need to spawn more daemons to handle the flow
+of new requests. The same applies if you want to handle high-availability
+scenarios: just start more Gnocchi daemons on independent servers.
+
+
+Understanding aggregation
+-------------------------
+
+The way data points are aggregated is configurable on a per-metric basis, using
+an archive policy.
+
+An archive policy defines which aggregations to compute and how many aggregates
+to keep. Gnocchi supports a wild number of aggregation methods, such as
+minimum, maximum, average, Nth percentile, standard deviation, etc. Those
+aggregations are computed over a period of time (called granularity) and are
+kept for a defined timespan. Aggregates are stored in a compressed format,
+ensuring the data points take as little space as possible.
```

Comment:
> Aggregates are stored in a compressed format, ensuring the data points take as little space as possible

this seems like an implementation detail. we should remove this or put it somewhere else

Reply:
removed